frontend/src/routes/_layout.tsx

```diff
@@ -0,0 +1,185 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { Outlet, createFileRoute, useNavigate } from "@tanstack/react-router";
+import {
+  AlarmCheck,
+  BookOpen,
+  ChevronDown,
+  ChevronsUpDown,
+  CircleUser,
+  Folder,
+  Globe,
+  Hammer,
+  HeartHandshake,
+  HelpCircle,
+  LogOut,
+  Search,
+  Send,
+  Settings,
+  Twitter
+} from "lucide-react";
+import { apiClient } from "~/api/client";
+import { useAuthUser } from "~/components/helper/use-auth-user";
+import { Logo } from "~/components/logo";
+import { Input } from "~/components/ui/input";
+import {
+  Menubar,
+  MenubarContent,
+  MenubarItem,
+  MenubarMenu,
+  MenubarTrigger
+} from "~/components/ui/menubar";
+import { userKeys } from "~/key-factories";
+import { cn } from "~/lib/utils";
+import { deleteCookie, getCookie } from "~/utils";
+
+export const Route = createFileRoute("/_layout")({
+  component: () => (
+    <div>
+      <Header />
+      <main>
+        <Outlet />
+      </main>
+      <Footer />
+    </div>
+  )
+});
+
+function Header() {
+  const query = useAuthUser();
+  const navigate = useNavigate();
+  const user = query.data?.data?.user;
+  const queryClient = useQueryClient();
+  const { data, isPending, mutate } = useMutation({
+    mutationFn: async () => {
+      // set csrf cookie token
+      await apiClient.GET("/api/csrf/");
+      const csrfToken = getCookie("csrftoken");
+      const { error } = await apiClient.DELETE("/api/auth/logout/", {
+        headers: {
+          "X-CSRFToken": csrfToken
+        }
+      });
+      if (error) {
+        return error;
+      }
+
+      queryClient.removeQueries({
+        queryKey: userKeys.authedUser
+      });
+      deleteCookie("csrftoken");
+      navigate({ to: "/login" });
+      return null;
+    }
+  });
+
+  if (!user) {
+    return null;
+  }
+  return (
+    <>
+      <header className="flex px-6 border-b border-opacity-65 border-border py-2 items-center bg-toogle t justify-between gap-4">
+        <Logo className="w-10 flex-none h-10 mr-8" />
```

Comment:
```suggestion
        <Link href="/dashboard">
          <Logo className="w-10 flex-none h-10 mr-8" />
        </Link>
```

This one is supposed to be wrapped in a link